cloud dm: support scaling migration jobs (#13249)
pingcap/docs: tidb-cloud/migrate-from-mysql-using-data-migration.md

## Scale a migration job

TiDB Cloud supports scaling up and down a migration job specification to meet your performance and cost requirements in different scenarios.

Different migration specifications have different performances. Your performance requirements might vary at different stages as well. For example, during a full data migration, you want the performance to be as fast as possible, so you choose a migration job with a large specification, for example 8 RCU. When the full migration is completed, the incremental migration does not require such a high performance, so you can scale down the job, for example, from 8 RCU to 2 RUC, to save cost.

When scaling a migration job, note the following:

- It takes about 5 to 10 minutes to scale a migration job.
- If the scaling fails, the job specification remains the same as before the scaling.

### Limitations

- You can only scale a migration job in the **Running** or **Paused** status.
- TiDB Cloud does not support scaling a migration job during the full data export stage.

> - Scaling a migration job will restart the task. If the task does not have a primary key, duplicate data might be inserted.
> During scaling, do not purge the binary log of the source database; otherwise, the job might fail because it cannot get the continuous binary log position. 

Author: ```suggestion
- Scaling a migration job will restart the task. If the task does not have a primary key, duplicate data might be inserted.
- During scaling, do not purge the binary log of the source database; otherwise, the job might fail because it cannot get the continuous binary log position. 

```

You can delete a migration job in any status.

If you encounter any problems during the migration, see [Migration errors and solutions](/tidb-cloud/tidb-cloud-dm-precheck-and-troubleshooting.md#migration-errors-and-solutions).

> ## Scale a migration job

Author: ```suggestion
## Scale a migration job specification
```

> TiDB Cloud supports scaling up and down a migration job specification to meet your performance and cost requirements in different scenarios.

Reviewer: ```suggestion
TiDB Cloud supports scaling up or down a migration job specification to meet your performance and cost requirements in different scenarios.
```

> Different migration specifications have different performances. Your performance requirements might vary at different stages as well. For example, during a full data migration, you want the performance to be as fast as possible, so you choose a migration job with a large specification, for example 8 RCU. When the full migration is completed, the incremental migration does not require such a high performance, so you can scale down the job specification, for example, from 8 RCU to 2 RUC, to save cost.

Reviewer: ```suggestion
Different migration specifications have different performances. Your performance requirements might vary at different stages as well. For example, during the existing data migration, you want the performance to be as fast as possible, so you choose a migration job with a large specification, such as 8 RCU. Once the existing data migration is completed, the incremental migration does not require such a high performance, so you can scale down the job specification, for example, from 8 RCU to 2 RUC, to save cost.
```

Reviewer: 这篇文档里其他几个地方，目前用的都是 existing data migration

> When scaling a migration job, note the following:

Author: ```suggestion
When scaling a migration job specification, note the following:
```

> - It takes about 5 minutes to scale a migration job.

Reviewer 2: ```suggestion
- It takes about 5 to 10 minutes to scale a migration job.
```
scale need to create a new node just like create job

> - It takes about 5 to 10 minutes to scale a migration job.

Author: ```suggestion
- It takes about 5 to 10 minutes to scale a migration job specification.
```

> - Resource specifications are calculated and billed by minute.

Author: ```suggestion
```

> - If the scaling fails, the job specification remains the same as before the scaling.

Reviewer: ```suggestion
- If the scaling fails, the job specification remains the same as it was before the scaling.
```

### Limitations

> - You can only scale a migration job in the **Running** or **Paused** status.

Author: ```suggestion
- You can only scale a migration job specification in the **Running** or **Paused** status.
```

> - You can only scale a migration job specification in the **Running** or **Paused** status.

Reviewer: ```suggestion
- You can only scale a migration job specification when the job is in the **Running** or **Paused** status.
```

> - TiDB Cloud does not support scaling a migration job specification during the full data export stage.

Reviewer: ```suggestion
- TiDB Cloud does not support scaling a migration job specification during the existing data export stage.
```

> - TiDB Cloud does not support scaling a migration job during the full data export stage.
> - Scaling a migration job will restart the task. If the task does not have a primary key, duplicate data might be inserted.

Author: ```suggestion
- TiDB Cloud does not support scaling a migration job specification during the full data export stage.
- Scaling a migration job specification will restart the task. If the task does not have a primary key, duplicate data might be inserted.
```

- Scaling a migration job will restart the task. If the task does not have a primary key, duplicate data might be inserted.

> 

Reviewer 3: ```suggestion
During the time of scaling, don't purge the binary log of source database, otherwise, the job may be failed which caused by can't get the continuous binary log position. 
```

> - Scaling a migration job specification will restart the task. If the task does not have a primary key, duplicate data might be inserted.

Reviewer: ```suggestion
- Scaling a migration job specification will restart the job. If a source table of the job does not have a primary key, duplicate data might be inserted.
```

> - During scaling, do not purge the binary log of the source database or increase `expire_logs_days` of the upstream database temporarily; otherwise, the job might fail because it cannot get the continuous binary log position. 

Reviewer: ```suggestion
- During scaling, do not purge the binary log of the source database or increase `expire_logs_days` of the upstream database temporarily. Otherwise, the job might fail because it cannot get the continuous binary log position. 
```

### Scaling procedure

1. Log in to the [TiDB Cloud console](https://tidbcloud.com/) and navigate to the [**Clusters**](https://tidbcloud.com/console/clusters) page of your project.

2. Click the name of your target cluster to go to its overview page, and then click **Data Migration** in the left navigation pane.

> 3. On the **Data Migration** page, locate the migration job you want to scale. In the **Action** column, click **Scale Up/Down**.

Reviewer: ```suggestion
3. On the **Data Migration** page, locate the migration job you want to scale. In the **Action** column, click **...** > **Scale Up/Down**.
```

Author: 我昨天验证，界面上并没有 ...

> 4. On the **Scale Up/Down** window, select the new specification you want to use, and then click **Submit**. You can view the new price of the specification at the bottom of the window.

Reviewer: ```suggestion
4. In the **Scale Up/Down** window, select the new specification you want to use, and then click **Submit**. You can view the new price of the specification at the bottom of the window.
```
